shap/_explanation.py

```diff
@@ -776,7 +782,7 @@ def _first_item(x):
     return None
 
 
-def _compute_shape(x):
+def _compute_shape(x) -> tuple[Union[int, None], ...]:
```

Comment:
A general question here: I found that you only added type hints for the return types not the inputs parameters, right? Is there a particular reason for this? I could imagine that it's easier iterate this way, but happy to understand this deeper.

Don't get me wrong, just want to understand this. No need to add type hints for input parameters in here.

Reply:
Good question, there's no particular reason just starting with this.

I was looking in a different PR about a good type hint for `X` - e.g. "ArrayLike", "slicable"; I think that's a larger issue so I thought I'd leave that for later for now

Reply:
sounds good, I am all for the approach to cut things down into manageable chunks.